sdk/core/azure-core/src/main/java/com/azure/core/credential/AzureSasCredential.java

```diff
@@ -53,7 +75,11 @@ public AzureSasCredential update(String signature) {
             throw logger.logExceptionAsError(new IllegalArgumentException("'signature' cannot be empty."));
         }
 
-        this.signature = signature;
+        this.signature = sanitizeSignature(signature, sanitationFunction);
         return this;
     }
+
+    private static String sanitizeSignature(String signature, Function<String, String> sanitationFunction) {
```

Comment:
Don't think this needs to be a static method. We can just use the instance method here and rely on the private final function as that should not change for each call to this method.